music21/metadata/__init__.py

```diff
@@ -96,161 +184,499 @@ class Metadata(base.Music21Object):
+        nsKeys with multiple values will appear multiple times in the list (rather
+        than appearing once, with a value that is a list of values).
+        The tuple's first element is either of the form 'namespace:name', or a
+        custom key (with no form at all).
+
+        >>> md = metadata.Metadata()
+        >>> md.add('composer', 'Jeff Bowen')
+        >>> md.add('librettist', 'Hunter Bell')
+        >>> md.add('title', 'Other World')
+        >>> md.addCustom('excerpt-start-measure', 1234)
+        >>> all = md.getAllNamedValues()
+        >>> all
+        [('marcrel:CMP', <music21.metadata.primitives.Contributor composer:Jeff Bowen>),
+        ('marcrel:LBT', <music21.metadata.primitives.Contributor librettist:Hunter Bell>),
+        ('dcterms:title', <music21.metadata.primitives.Text Other World>),
+        ('excerpt-start-measure', <music21.metadata.primitives.Text 1234>)]
+        >>> allNonContributors = md.getAllNamedValues(skipContributors=True)
+        >>> allNonContributors
+        [('dcterms:title', <music21.metadata.primitives.Text Other World>),
+        ('excerpt-start-measure', <music21.metadata.primitives.Text 1234>)]
+        '''
+        allOut: t.List[t.Tuple[str, ValueType]] = []
+
+        valueList: t.List[ValueType]
+        for nsKey, valueList in self._metadata.items():
+            if skipContributors and self._isContributorNSKey(nsKey):
+                continue
+
+            value: ValueType
+            for value in valueList:
+                allOut.append((nsKey, value))
+
+        return allOut
+
+    def getAllContributorNamedValues(self) -> t.List[t.Tuple[str, ValueType]]:
+        '''
+        Returns all contributors stored in this metadata as a list of (nsKey, value) tuples.
+        The tuple's first element will be of the form 'namespace:name'.
+
+        >>> md = metadata.Metadata()
+        >>> md['title'] = ['Dimitrij', 'False Dmitry']
+        >>> md.composer = 'Antonín Dvořák'
+        >>> md.opusNumber = 64
+        >>> md.add('librettist', 'Marie Červinková-Riegrová')
+        >>> md['otherContributor'] = (metadata.Contributor(role='based on plot by',
+        ...                               name ='Ferdinand Mikovec'),
+        ...                           metadata.Contributor(role='original partial plot by',
+        ...                               name='Friedrich Schiller'))
+        >>> md.addCustom('composer', 'Not a contributor')
+        >>> allContributors = md.getAllContributorNamedValues()
+        >>> allContributors
+        [('marcrel:CMP', <music21.metadata.primitives.Contributor composer:Antonín Dvořák>),
+        ('marcrel:LBT',
+        <music21.metadata.primitives.Contributor librettist:Marie Červinková-Riegrová>),
+        ('marcrel:CTB',
+        <music21.metadata.primitives.Contributor based on plot by:Ferdinand Mikovec>),
+        ('marcrel:CTB',
+        <music21.metadata.primitives.Contributor original partial plot by:Friedrich Schiller>)]
```

Comment:
indent particularly important here.  Wrap text to show parallelism (including with the CMP line)

Reply:
Done.